packages/libp2p/src/identify/index.ts

```diff
@@ -60,6 +66,22 @@ export const multicodecs = {
 
 export const Message = { Identify }
 
+export interface IdentifyService {
+  /**
+   * due to the default limits on inbound/outbound streams for this protocol,
+   * invoking this method when runOnConnectionOpen is true can lead to unpredictable results
+   * as streams may be closed by the local or the remote node.
+   * Please use with caution. If you find yourself needing to call this method to discover other peers that support your protocol,
+   * you may be better off configuring a topology to be notified instead.
+   *
+   * @param connection
+   * @param options
+   * @returns
+   */
+  identify: (connection: Connection, options: AbortOptions) => Promise<IdentifyResult>
+
+}
+
 export function identifyService (init: IdentifyServiceInit = {}): (components: IdentifyServiceComponents) => DefaultIdentifyService {
```

Comment:
This should return the interface

```suggestion
export function identifyService (init: IdentifyServiceInit = {}): (components: IdentifyServiceComponents) => IdentifyService {
```

Reply:
This has [implications beyond the scope of this PR](https://github.com/libp2p/js-libp2p/actions/runs/5405601445/jobs/9821368599?pr=1784#step:4:316) which can be refactored in a follow up PR.